caikit_nlp/modules/token_classification/filtered_span_classification.py

```diff
@@ -111,53 +132,97 @@ def run(
+        based on score threshold for labels that are to be outputted
+
+        Args:
+            text_stream: Iterable[str]
+                Text stream to run classification on
+            threshold: float
+                (Optional) Threshold based on which to return score results
 
+        Returns:
+            Iterable[StreamingTokenClassificationResult]
+        """
         # TODO: For optimization implement window based approach.
+        if threshold is None:
+            threshold = self.default_threshold
 
+        offset = 0
         for span_output in self._stream_span_output(text_stream):
-            classification_result = self.sequence_classifier.run(span_output.text)
+            text_len = span_output.end - span_output.start
+            classification_result = self.classifier.run(span_output.text)
             for classification in classification_result.results:
-                label = classification.label
-                # TODO: Figure out how we want to handle thresholding for streaming
-                # since with streaming, if the sentence we are processing
-                # is below threshold, do we want to skip yielding that response and try to get
-                # another one ? What if none cross / passes threshold.
-                yield TokenClassification(
-                    start=span_output.start,
-                    end=span_output.span.end,
-                    word=span_output.span.text,
-                    entity=label,
-                    score=classification.score,
-                )
-                # TODO: What would be final output DM for streaming.
+                # TODO: refactor common code between .run and here
+                if self.classification_task == TextClassificationTask:
+                    label = classification.label
+                    start = span_output.start
+                    end = span_output.end
+                    word = span_output.text
+                else:
+                    label = classification.entity
+                    start = classification.start
+                    end = classification.end
+                    word = classification.word
+                if classification.score >= threshold:
+                    if not self.labels_to_output or (
+                        self.labels_to_output and label in self.labels_to_output
+                    ):
+                        # TODO: update output DM
+                        # Need to add offset to track actual place of spans within a stream,
+                        # as the span splitting will be expected to stream and detect spans
+                        yield TokenClassification(
+                            start=start + offset,
+                            end=end + offset,
+                            word=word,
+                            entity=label,
+                            score=classification.score,
+                        )
+            offset += text_len
```

Comment:
hmm. can we just use `span_output.end` directly instead of calculating based on length?

Reply:
I suppose that's true here since the implication is that `start` would be `0`